### PR TITLE
Add unmock function

### PR DESCRIPTION
**Pull request by Chemaclass**
## 📚 Description

Currently, you can mock a particular function with custom logic, but there is no way to rollback the mock for other tests. It would be useful to be able to remove/rollback the mocked behaviour. 

## 🔖 Changes

- Add unmock function

All mocks will be removed/rollback to their original behaviour after each test. 
This means that the mock live scope is tied to the test scope

## ✅ To-do list

- [x] I updated the `CHANGELOG.md` to reflect the new feature or fix
- [x] I updated the documentation to reflect the changes

### CODE REVIEW

**Review by JesusValera — APPROVED** [review]

File: tests/unit/test_doubles_test.sh [hunk 1]
```diff
@@ -24,6 +24,22 @@ EOF
...
+
+  mock ps<<EOF
+PID TTY          TIME CMD
+13525 pts/7    00:00:01 bash
+24162 pts/7    00:00:00 ps
+EOF
+
+  assert_equals "3" "$(helper::trim "$(code)")"
+  unmock ps
+  assert_not_equals "3" "$(helper::trim "$(code)")"
```
Comment (khru) [inline tests/unit/test_doubles_test.sh:40]: How could return 3 on your OS?

Comment (khru) [reply tests/unit/test_doubles_test.sh:40]: this has to still mock 
image

**Review by khru — CHANGES_REQUESTED** [review]

File: tests/unit/test_doubles_test.sh (hunk 1 above)
Comment (khru) [reply tests/unit/test_doubles_test.sh:40]: an the test it's not telling you what behavior it's expected

Comment (Chemaclass) [reply tests/unit/test_doubles_test.sh]: You're right. I will use a different approach.

File: src/runner.sh [hunk 2]
```diff
@@ -127,6 +127,7 @@ function runner::run_test() {
     "$function_name" "$data" 2>&1 1>&3
 
     runner::run_tear_down
+    runner::clean_mocks
```
Comment (khru) [inline src/runner.sh:130]: Where are the test?

File: src/runner.sh [hunk 3]
```diff
@@ -189,6 +190,12 @@ function runner::run_tear_down() {
   helper::execute_function_if_exists 'tear_down'
 }
 
+function runner::clean_mocks() {
```
Comment (khru) [inline src/runner.sh:193]: Where are the test

**Review by khru — CHANGES_REQUESTED** [review]

File: src/runner.sh (hunk 2 above)
Comment (Chemaclass) [reply src/runner.sh:130]: I added them to `tests/acceptance/mock_test.sh`

Comment (khru) [reply src/runner.sh:130]: we are also missing unit test of the function, no?

Comment (khru) [inline src/runner.sh:130]: In other frameworks, this function is call `clear_mocks`
image

File: tests/acceptance/mock_test.sh [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  assert_is_mock ls
+
+  spy ps
+  assert_is_mock ps
+}
+
+function test_runner_clean_mocks_2() {
+  assert_is_not_mock ls
+  assert_is_not_mock ps
+}
```
Comment (khru) [inline tests/acceptance/mock_test.sh:17]: These test are not acceptance test, and the behavior you are trying to test it's not clear

File: tests/unit/test_doubles_test.sh [hunk 5]
```diff
@@ -24,6 +24,25 @@ EOF
...
+13525 pts/7    00:00:01 bash
+24162 pts/7    00:00:00 ps
+EOF
+
+  assert_is_mock ps
+  unmock ps
+  assert_is_not_mock ps
+}
+
+function test_successful_unmock_spy() {
```
Comment (khru) [inline tests/unit/test_doubles_test.sh:39]: where are the unsuccessful test for this?

**Review by khru — CHANGES_REQUESTED** [review]

**Comment by Tito-Kati** [comment]
@Chemaclass this is more complex than we anticipated, do you mind if we leave it in draft until we can get together? Besides, I think it's more priority to refactor the acceptance tests and finish the snapshots.

File: tests/unit/test_doubles_test.sh (hunk 5 above)
Comment (Chemaclass) [reply tests/unit/test_doubles_test.sh:39]: Added

File: src/runner.sh (hunk 2 above)
Comment (Chemaclass) [reply src/runner.sh]: Could you give me a hint how would you write unit tests for `runner::clear_mocks()`?

Comment (Chemaclass) [reply src/runner.sh]: Done

**Review by CosmeValera — APPROVED** [review]

**Comment by Chemaclass** [comment]
Close because inactivity. I will open it again to seek for new feedback

**Pull request closed without merging**